File: ChessProject-Java/src/main/java/com/solarwindsmsp/chess/IPiece.java
package com.solarwindsmsp.chess;

public interface IPiece {

    ChessBoard getChesssBoard();
    void setChessBoard(ChessBoard chessBoard);

    int getXCoordinate();
    void setXCoordinate(int value);

    int getYCoordinate();
    void setYCoordinate(int value);

    void Move(MovementType movementType, int newX, int newY);

    PieceColor getPieceColor();

}

File: ChessProject-Csharp/src/Interfaces/IPiece.cs
using SolarWinds.MSP.Chess.Models;
using System;

namespace SolarWinds.MSP.Chess.Interfaces
{
    interface IPiece
    {
        int XCoordinate { get; set; }
        int YCoordinate { get; set; }
        PieceColor PieceColor { get; }
        int MaxForColour { get; }

        void Move(MovementType movementType, int newX, int newY);
    }
}
